### PR TITLE
Extend query documentation

### PR DESCRIPTION
**Pull request by ThirVondukr**
Extends query documentation, adding examples with using decorator, arguments and `strawberry.argument`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/general/queries.md [hunk 1]
```diff
@@ -44,3 +44,57 @@ schema = strawberry.Schema(query=Query)
...
+class Query:
+    @strawberry.field
+    def fruit(self, startswith: str) -> str | None:
+        for fruit in FRUITS:
+            if fruit.startswith(startswith):
+                return fruit
+        return None
+```
+
+Additional metadata can be added to arguments like a different name and
```
Comment (DoctorJohn) [inline docs/general/queries.md:81]: Just a suggestion on how I would have worded it. Feel free to ignore it :)

```suggestion
Additional metadata can be added to arguments, for example a custom name and
```

**Review by DoctorJohn — APPROVED** [review]
Thanks for the PR @ThirVondukr, LGTM :)

File: docs/general/queries.md (hunk 1 above)
Comment (ThirVondukr) [reply docs/general/queries.md:81]: I like it more

File: docs/general/queries.md [hunk 2]
```diff
@@ -44,3 +44,57 @@ schema = strawberry.Schema(query=Query)
...
+class Query:
+    @strawberry.field
+    def fruits(
+        self,
+        is_tasty: Annotated[
+            bool | None,
+            strawberry.argument(
+                description="Filters out fruits by whenever they're tasty or not",
+                deprecation_reason="isTasty argument is deprecated, "
+                "use fruits(taste:SWEET) instead",
```
Comment (NafieAlhilaly) [inline docs/general/queries.md:96]: How about 🤔
```suggestion
                deprecation_reason="isTasty argument is deprecated, use fruits(taste:SWEET) instead",
```

----
```python
@strawberry.type
class Query:
    @strawberry.field
    def fruits(
        self,
        is_tasty: Annotated[
            bool | None,
            strawberry.argument(
                description="Filters out fruits by whenever they're tasty or not",
                deprecation_reason="isTasty argument is deprecated, "
                "use fruits(taste:SWEET) instead",
            ),
        ] = None,
    ) -> list[str]: ...
```

----
```python
@strawberry.type
class Query:
    @strawberry.field
    def fruits(
        self,
        is_tasty: Annotated[
            bool | None,
            strawberry.argument(
                description="Filters out fruits by whenever they're tasty or not",
                deprecation_reason="isTasty argument is deprecated, use fruits(taste:SWEET) instead",
            ),
        ] = None,
    ) -> list[str]: ...
```

Comment (ThirVondukr) [reply docs/general/queries.md:96]: This it a bit too long (101) characters, it's hard to read in git diffs, and black by default recommends 88-80 character limit

**Pull request merged**